src/session.ts

```diff
@@ -280,11 +282,27 @@ export class SessionManager implements Middleware {
                 });
         }
 
+        // If powershell.powerShellDefaultVersion specified, attempt to find the PowerShell exe path
+        // of the version specified by the setting.
+        if ((this.sessionStatus === SessionStatus.NeverStarted) && this.sessionSettings.powerShellDefaultVersion) {
+            const powerShellExePaths = getAvailablePowerShellExes(this.platformDetails, this.sessionSettings);
+            const powerShellDefaultVersion =
+                powerShellExePaths.find((item) => item.versionName === this.sessionSettings.powerShellDefaultVersion);
+
+            if (powerShellDefaultVersion) {
+                powerShellExePath = powerShellDefaultVersion.exePath;
+            } else {
+                this.log.writeWarning(
+                    `Could not find powerShellDefaultVersion: '${this.sessionSettings.powerShellDefaultVersion}'`);
+            }
+        }
+
         // Is there a setting override for the PowerShell path?
-        let powerShellExePath =
-            (this.sessionSettings.powerShellExePath ||
-                this.sessionSettings.developer.powerShellExePath ||
-                "").trim();
+        powerShellExePath =
+            (powerShellExePath ||
+             this.sessionSettings.powerShellExePath ||
+             this.sessionSettings.developer.powerShellExePath ||
```

Comment:
this is for backcompat?

Reply:
The check for `this.sessionSettings.developer.powerShellExePath` is for back compat.  For 2.0, I say we remove the setting and this fallback.

Reply:
I'm assuming it's a fallback? Should the developer path come before the other one, or is the intent to need to blank out the normal path to get the dev one?

Reply:
Yeah back in the day it was called `this.sessionSettings.developer.powerShellExePath` that was changed to `this.sessionSettings.powerShellExePath`. Now Keith is changing that. Since VSCode doesn't notify when a setting has become deprecated (AFAIK) we support them as fallback until we want to remove support.

Keith, I totally agree v2 should remove this.